client/components/vertical-nav/item/enhanced.jsx

```diff
@@ -45,10 +47,12 @@ const VerticalNavItemEnhanced = ( {
 	onClick,
 	text,
 } ) => {
+	const classes = classnames( 'vertical-nav-item-enhanced', className );
+
 	return (
 		<VerticalNavItem
 			// eslint-disable-next-line wpcalypso/jsx-classname-namespace
-			className="vertical-nav-item-enhanced"
+			className={ classes }
```

Comment:
Rather than pass this in, I think it makes more sense to implement this using a `disabled` prop for `VerticalNavitem` -- that way we can consistently style and manage the UX for a disabled menu item. We could then have the calculation for `classes` above be something more like:
```
const classes = classnames( 'vertical-nav-item-enhanced', { disabled } );
```
And have the default disabled CSS be driven by `&.disabled` selectors in the CSS (in addition to disabling the click handling etc for the`<a>` element that the component generates).

In general, we try to aim for `is` or `has` prefixes for boolean flags, but I think that `disabled` is such a common attribute for HTML it's probably simpler to stick with that.

Reply:
I had the same concern, will do this. 

I would prefer keeping the `className` prop for `VerticalNavItemEnhanced` as well though.

Reply:
Done. I have kept the `className` prop in `VerticalNavItemEnhanced`, doesn't hurt to make it configurable.